internal/manifest/iso_tree_ostree_payload.go

```diff
@@ -0,0 +1,64 @@
+package manifest
+
+import "github.com/osbuild/osbuild-composer/internal/osbuild2"
+
+const ostreeRepoPath = "/ostree/repo"
+
+type osTreeISOTreePayload struct {
+	osTreeURL    string
+	osTreeRef    string
+	osTreeCommit string
+	osName       string
+}
+
+func NewOSTreeISOTreePayload(osTreeURL, osTreeRef, osTreeCommit, osName string) ISOTreePayload {
+	return osTreeISOTreePayload{
+		osTreeURL:    osTreeURL,
+		osTreeRef:    osTreeRef,
+		osTreeCommit: osTreeCommit,
+		osName:       osName,
+	}
+}
+
+func (o osTreeISOTreePayload) getBuildPackages() []string {
+	return []string{"rpm-ostree"}
+}
+
+func (o osTreeISOTreePayload) getImageURL() string {
+	return ""
+}
+
+func (o osTreeISOTreePayload) getOSTreeURL() string {
+	return o.osTreeURL
+}
+
+func (o osTreeISOTreePayload) getOSTreeRef() string {
+	return o.osTreeRef
+}
+
+func (o osTreeISOTreePayload) getOSTreeURLForKickstart() string {
+	return makeISORootPath(ostreeRepoPath)
+}
+
+func (o osTreeISOTreePayload) getOSTreeCommits() []osTreeCommit {
+	return []osTreeCommit{
+		{
+			checksum: o.osTreeCommit,
+			url:      o.osTreeURL,
+		},
+	}
+}
+
+func (o osTreeISOTreePayload) getOSName() string {
+	return o.osName
+}
+
+func (o osTreeISOTreePayload) getPayloadStages() []*osbuild2.Stage {
```

Comment:
So far all stages were generated only in the pipeline types, not in helpers, if we go this route (though I suggested above an alternative), would it make sense to just return the `osTreeCommit` and `osTreeRef` properties from this function, and instead do the stage stuff in the pipeline generation function? That way we could reuse this even if we change how the stages are put together.